homeassistant/helpers/translation.py

```diff
@@ -323,3 +337,111 @@ async def async_get_translations(
     for entry in cached:
         result.update(entry)
     return result
+
+
+async def async_load_translations(
+    hass: HomeAssistant,
+    language: str,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> None:
+    """Prime backend translation cache.
+
+    If integrations are not specified, translation cache is primed for all loaded integrations.
+    """
+    lock = hass.data.setdefault(TRANSLATION_LOAD_LOCK, asyncio.Lock())
+
+    components = _async_get_components_to_load(hass, category, integrations)
+
+    async with lock:
+        if TRANSLATION_FLATTEN_CACHE in hass.data:
+            cache = hass.data[TRANSLATION_FLATTEN_CACHE]
+        else:
+            cache = hass.data[TRANSLATION_FLATTEN_CACHE] = _TranslationCache(hass)
+        await cache.async_load(language, components)
+
+
+@callback
+def async_get_cached_translations(
+    hass: HomeAssistant,
+    language: str,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> dict[str, Any]:
+    """Return all cached backend translations.
+
+    If integrations specified, return translations for them.
+    Otherwise default to loaded integrations.
+    """
+    components = _async_get_components_to_load(hass, category, integrations)
+
+    if TRANSLATION_FLATTEN_CACHE not in hass.data:
+        return {}
+
+    result = {}
+
+    cache = hass.data[TRANSLATION_FLATTEN_CACHE]
+    cached = cache.get_cached(language, category, components)
+    for entry in cached:
+        result.update(entry)
+
+    return result
+
+
+@callback
+def _async_get_components_to_load(
+    hass: HomeAssistant,
+    category: str,
+    integrations: Iterable[str] | None = None,
+) -> set[str]:
+    if integrations is not None:
+        components = set(integrations)
+    elif category in ("state", "entity_component", "services"):
+        components = set(hass.config.components)
+    else:
+        # Only 'state' supports merging, so remove platforms from selection
+        components = {
+            component for component in hass.config.components if "." not in component
+        }
+    return components
+
+
+async def async_load_state_translations_to_cache(
```

Comment:
This should not be public
```suggestion
async def _async_load_state_translations_to_cache(
```